src/lib/storage.js

```diff
@@ -507,7 +509,14 @@ class Storage implements IStorageHandler {
           if (err) {
             return callback(err);
           }
-          return callback(null, packageJsonLocal, upLinksErrors);
+          async.map(
+            self.filters,
+            (filter, cb) => filter.filter_metadata(packageJsonLocal, cb),
```

Comment:
Perhaps we should rely on Promises from the very beginning instead callbacks

Reply:
Very much agreed. I'll try to clean this up

Reply:
Fixed